### PR TITLE
cppcheck: fix some reports

### PR DESCRIPTION
**Pull request by serval2412**
[core/winnt/src/TWinNTSystem.cxx:996]: (error) Invalid number of character '(' when these macros are defined: 'ROOTPREFIX'.
[html/src/TDocDirective.cxx:316]: (error) Invalid number of character '{' when these macros are defined: 'R__BEPAEPSTLICHERALSDERPAPST'.
[math/mathcore/src/Delaunay2D.cxx:30]: (error) Invalid number of character '{' when these macros are defined: 'HAS_CGAL'.

[graf2d/qt/src/TGQt.cxx:2817]: (error) Array 'shape[15]' accessed at index 15, which is out of bounds.
[graf2d/qt/src/TGQt.cxx:2818]: (error) Array 'shape[15]' accessed at index 16, which is out of bounds.
[graf2d/qt/src/TGQt.cxx:2841]: (error) Array 'shape[15]' accessed at index 15, which is out of bounds.
[net/http/civetweb/handle_form.inl:627]: (style) Array index 'd' is used before limits check.

[roofit/roostats/src/MetropolisHastings.cxx:153]: (style) Condition 'i<1000' is always true

[graf2d/qt/src/TQtClientGuard.cxx:289]: (warning) Assert statement modifies 'thisPix'.

[roofit/roofitcore/src/BidirMMapPipe.cxx:906]: (style) Throwing a copy of the caught exception instead of rethrowing the original exception.
[roofit/roofitcore/src/BidirMMapPipe.cxx:972]: (style) Throwing a copy of the caught exception instead of rethrowing the original exception.
[roofit/roofitcore/src/BidirMMapPipe.cxx:1394]: (style) Throwing a copy of the caught exception instead of rethrowing the original exception.
[roofit/roofitcore/src/BidirMMapPipe.cxx:1429]: (style) Throwing a copy of the caught exception instead of rethrowing the original exception.

[graf3d/eve/src/TEveElement.cxx:1745]: (style) Exception should be caught by reference.
[roofit/histfactory/src/hist2workspace.cxx:76]: (style) Exception should be caught by reference.
[roofit/histfactory/src/hist2workspace.cxx:97]: (style) Exception should be caught by reference.
[roofit/roofitcore/src/RooClassFactory.cxx:786]: (style) Exception should be caught by reference.
[roofit/roofitcore/src/RooFactoryWSTool.cxx:413]: (style) Exception should be caught by reference.
[roofit/roofitcore/src/RooFactoryWSTool.cxx:484]: (style) Exception should be caught by reference.
[roofit/roofitcore/src/RooFactoryWSTool.cxx:526]: (style) Exception should be caught by reference.
[roofit/roofitcore/src/RooFactoryWSTool.cxx:568]: (style) Exception should be caught by reference.
[roofit/roofitcore/src/RooFactoryWSTool.cxx:588]: (style) Exception should be caught by reference.
[roofit/roofitcore/src/RooFactoryWSTool.cxx:629]: (style) Exception should be caught by reference.
[roofit/roofitcore/src/RooFactoryWSTool.cxx:642]: (style) Exception should be caught by reference.
[roofit/roofitcore/src/RooFactoryWSTool.cxx:681]: (style) Exception should be caught by reference.
[roofit/roofitcore/src/RooFactoryWSTool.cxx:859]: (style) Exception should be caught by reference.
[tmva/tmva/src/CostComplexityPruneTool.cxx:131]: (style) Exception should be caught by reference.
[tmva/tmva/src/CostComplexityPruneTool.cxx:142]: (style) Exception should be caught by reference.
[tmva/tmva/src/DataSet.cxx:173]: (style) Exception should be caught by reference.
[tmva/tmva/src/Reader.cxx:636]: (style) Exception should be caught by reference.
[tmva/tmva/src/Reader.cxx:702]: (style) Exception should be caught by reference.
[tmva/tmva/src/VariableGaussTransform.cxx:751]: (style) Exception should be caught by reference.

[io/sql/src/TSQLStructure.cxx:1931] -> [io/sql/src/TSQLStructure.cxx:1933]: (warning) Identical condition 'len==0', second condition is always false

### CODE REVIEW

**Comment by amadio** [comment]
Could you please separate the commits by type of fix?

File: roofit/roofitcore/src/BidirMMapPipe.cxx [hunk 1]
```diff
@@ -903,7 +903,7 @@ BidirMMapPipe::BidirMMapPipe(bool useExceptions, bool useSocketpair) :
             delete s_pagepool;
             s_pagepool = 0;
         }
-        throw e;
+        throw;
```
Comment (amadio) [inline roofit/roofitcore/src/BidirMMapPipe.cxx]: Won't deleting this lead to unused variable warnings for `e`?

**Comment by amadio** [comment]
BTW, thanks a lot for the fixes!

**Comment by serval2412** [comment]
Amadio: I removed const and variable name in catches when just rethrowing the initial exception.
About splitting the patch, I don't know how to do this simply without redoing the whole stuff.

**Comment by amadio** [comment]
@serval2412 Thank you for updating the PR. I can split the changes myself, but if you prefer to try it, I can explain how to do it. Let me know what you prefer.

**Comment by serval2412** [comment]
If it doesn't bother you too much to split, i'd prefer you do it :-)
Indeed, I don't know git well. For example, to update my local repo once a patch has been merged, I must Google the commands to know how to do this. Last time, I didn't even succeed and finally remove my local repo and fork "root" again.
The basics of git are quite clear but between rebase, fetch, pull, branches, cherry-pick, I'm a bit lost.

**Comment by amadio** [comment]
@serval2412 Would you mind rebasing and pushing, so we can test with latest master? Thanks!

**Comment by serval2412** [comment]
@amadio : I did this:
git fetch upstream
git merge upstream/master
git push
is it ok?

**Comment by amadio** [comment]
No, please don't use merge, but rebase instead. `git rebase upstream/master`

**Comment by serval2412** [comment]
Hmm, I already did what I quoted in my last comment.
Now if I did:
git rebase upstream/master
->
Applying: cppcheck: fix some reports
Applying: fix warnings for catch
julien@debian:~/projects/root$ git push
Username for 'https://github.com': serval2412
Password for 'https://serval2412@github.com': 
To https://github.com/serval2412/root
 ! [rejected]              master -> master (non-fast-forward)
error: failed to push some refs to 'https://github.com/serval2412/root'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
What should I do then?

**Comment by amadio** [comment]
I think it's easier for me to just push the changes and merge as a separate PR (you will still be the author).
But just in case, when you rebase, you have to do `push --force`. This is now PR #1565. Thanks for the contribution!

**Pull request closed without merging**